jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketUpgradeHandler.java

```diff
@@ -83,19 +84,24 @@ public class WebSocketUpgradeHandler extends Handler.Wrapper
      * @see #configure(Consumer)
      */
     public static WebSocketUpgradeHandler from(Server server, ContextHandler context)
+    {
+        return from(server, context, WebSocketUpgradeHandler::new);
+    }
+
+    public static WebSocketUpgradeHandler from(Server server, ContextHandler context, Function<ServerWebSocketContainer, WebSocketUpgradeHandler> factory)
```

Comment:
Do we really need the static `from` methods on the `WebSocketUpgradeHandler`?

It would be more natural to do `new WebSocketUpgradeHandler(ServerWebSocketContainer.from(server, context))`, so a user could directly extend the handler rather than pass in a factory.

We would have to deprecate this `from` and add a static `from` onto `ServerWebSocketContainer`.  It is after all the container that we do not wish to duplicate and we could have multiple handlers without a problem... so long as they share the container.